Packet++/header/HttpLayer.h

```diff
@@ -207,201 +207,217 @@ namespace pcpp
-
-	class HttpResponseFirstLine;
-
+	// -------- Class HttpResponseStatusCode -----------------
 
 	/**
-	 * @class HttpResponseLayer
-	 * Represents an HTTP response header and inherits all basic functionality of HttpMessage and TextBasedProtocolMessage.
-	 * The functionality that is added for this class is the HTTP first line concept. An HTTP response has the following first line:
-	 * <i>200 OK HTTP/1.1</i>
-	 * Since it's not an "ordinary" HTTP field, it requires a special treatment and gets a class of it's own: HttpResponseFirstLine.
-	 * Unlike most L2-4 protocols, an HTTP response header can spread over more than 1 packet. PcapPlusPlus currently doesn't support a header
-	 * that is spread over more than 1 packet so in such cases: 1) only the first packet will be parsed as HttpResponseLayer (the other packets
-	 * won't be recognized as HttpResponseLayer) and 2) the HTTP header for the first packet won't be complete (as it continues in the following
-	 * packets), this why PcapPlusPlus can indicate that HTTP response header is complete or not (doesn't end with "\r\n\r\n" or "\n\n") using
-	 * HttpMessage#isHeaderComplete()
+	 * Class for the enum of HTTP response status codes
 	 */
-	class HttpResponseLayer : public HttpMessage
+	class HttpResponseStatusCode
 	{
-		friend class HttpResponseFirstLine;
 	public:
-		/**
-		 * Enum for HTTP response status codes
-		 */
-		enum HttpResponseStatusCode
+		enum Value: int
 		{
 			/** 100 Continue*/
-			Http100Continue,
+			Http100Continue = 100,
 			/** 101 Switching Protocols*/
-			Http101SwitchingProtocols,
+			Http101SwitchingProtocols = 101,
 			/** 102 Processing */
-			Http102Processing,
+			Http102Processing = 102,
+			/** 103 Early Hints */
+			Http103EarlyHints = 103,
+			/** 104-199 Unassigned */
+
 			/** 200 OK */
-			Http200OK,
+			Http200OK= 200,
 			/** 201 Created */
-			Http201Created,
+			Http201Created = 201,
 			/** 202 Accepted */
-			Http202Accepted,
+			Http202Accepted = 202,
 			/** 203 Non-Authoritative Information */
-			Http203NonAuthoritativeInformation,
+			Http203NonAuthoritativeInformation = 203,
 			/** 204 No Content*/
-			Http204NoContent,
+			Http204NoContent = 204,
 			/** 205 Reset Content*/
-			Http205ResetContent,
+			Http205ResetContent = 205,
 			/** 206 Partial Content */
-			Http206PartialContent,
+			Http206PartialContent = 206,
 			/** 207 Multi-Status */
-			Http207MultiStatus,
+			Http207MultiStatus = 207,
 			/** 208 Already Reported */
-			Http208AlreadyReported,
+			Http208AlreadyReported = 208,
+			/** 229-225 Unassigned */
 			/** 226 IM Used */
-			Http226IMUsed,
+			Http226IMUsed = 226,
+			/** 227-299 Unassigned */
+			
 			/** 300 Multiple Choices */
-			Http300MultipleChoices,
+			Http300MultipleChoices = 300,
 			/** 301 Moved Permanently */
-			Http301MovedPermanently,
-			/** 302 (various messages) */
-			Http302,
+			Http301MovedPermanently = 301,
+			/** 302 Found */
+			Http302Found = 302,
```

Comment:
previously `302` could also mean `Moved Temporarily`, do we want to keep supporting it?

Reply:
@seladb  answer about all comments for adding and removing code in the list:
as @egecetin  mentioned, those values are outdated, according to https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml, so I just updated them according to the latest spec (2022)
The question is should we support the old code or not? IMO, we should avoid supporting old specs, since it takes more effort to maintain and increases the complexity of the code base.
We probably can make this version a breaking change version?

Reply:
I think the unofficial ones should also be supported too as @seladb mentioned because some of them widely used. @tigercosmos so there is no need removing old codes only adding missing ones and apply changes enough. But the only problem what will we do when there is a conflict for a status code message (if there are some codes have this situation). Since they are unofficial, different apps can use same code for different messages. What do you think about this @seladb? Should all messages be displayed somehow (maybe comma separated etc), display only most popular one or display a generic message like "various messages" or ...?

Reply:
The essence of PcapPlusPlus is parsing packets. People use it to parse different kinds of packets, some of which are very old and contain outdated data, such as old HTTP messages. That's why I think we should support outdated/unofficial HTTP data.
It's true that applications can use HTTP status codes for any purpose, even if that purpose doesn't meet the official specifications. However, for most status codes there is only one usage (classic examples are `200 OK`, `404 Not Found`, `401 Unauthorized`, `500 Internal Server Error`). If there are cases where there is more than one usage, we can probably say "<Official usage> or other". Please let me know what you think

Reply:
I agree we should support any status code.
actually, even for unknown status codes (`1xx`-`5xx`, exclusive other else, e.g. `999`), I think we should also support it, e.g. `599 My Customized`, because users may want to analyze their internal usage packets.

the current behavior for parsing a packet limits the possibility of the user's customized status code.
how about this, we can add a new `std::string m_StatusCodeCustomizedDefination` in `HttpResponseFirstLine`, to allow the user's customized code.
- if the code is `200`, and it's already in our `HttpResponseFirstLine`, and the definition in the packet is the same as our definition, so  `m_StatusCodeCustomizedDefination` is empty
- if the code is `599`, and the user defines it's `My Customized`, then `m_StatusCodeCustomizedDefination`  is `My Customized`.
- if the code is `302`, it's `Found` in the spec, but the user defines it as `Hello World`,  then `m_StatusCodeCustomizedDefination`  is `Hello World`. 

later on, whenever we need to print or return the definition of the status code, we first see if `m_StatusCodeCustomizedDefination` is defined; otherwise, we return the default definition.

Reply:
`HttpResponseFirstLine` is probably not a great place to put this definition because it has a private constructor and cannot be modified by the user. Besides, it seems that customized codes should be part of `HttpResponseStatusCode`.

In order to keep this PR simple and focused, I'd suggest supporting the existing status codes and maybe add customized code later. Please let me know what you think

Reply:
sure, let's do it in another PR